pressbooks/pressbooks-multi-institution
# feat: institutional manager role

This PR address #10 and #39

This would require to open a PR on [pressbooks-networkanalytics](https://github.com/pressbooks/pressbooks-network-analytics/pull/331) and https://github.com/pressbooks/pressbooks/pull/3602

## What does this PR do.

* Add/remove institutional managers (network managers) when they are added/removed in the institutions form (using a hook)
* Hide unwanted menu items for Institutional manager
* Disable navigation on all pages by default and allow access to certain pages only
* Expose a useful function to check if the current user is an institutional manager and returns the instutition ID or 0 if not `get_institution_by_manager` instead of is_institutional_manager
* Exposes useful filters that can be used on views or queries to filter the data

```
pb_institutional_books
pb_institution
pb_institutional_managers
pb_institutional_users
```

For instance:

https://github.com/pressbooks/pressbooks-network-analytics/pull/331/files#diff-c20e4732fda4f9966fce7b1d6beb0c25c6017c85c546d62a2b0b027609b997a6R129

https://github.com/pressbooks/pressbooks/pull/3602/files#diff-29ab3f3c40495d554c34b2b53d32985b8d0ed7ea485c6d86b691bb70d27ca13bR607

### TODO
* Handle main site menu and top bar
* Tests

### How to test as (developer)
1. Checkout this branch
3. Checkout feat/filters-hooks in networkanalytics
4. Assign users as institutional managers on institution form
5. Login as institutional manager and observe the UI and options allowed
6. Try to visit any super admin url that an institutional manager should not be able to see
7. As a Super admin review the new institution managers are not displayed as network managers on Network Setup

### How to test as QA (early tester) 
1. Use https://oscardev.pressbooks.network/
2. Assign users as institutional managers on institution form
3. Login as one of the users you added in the previous step and observe if the menus are limited to the options available for institutional managers (Books, Users, Stats)
4. You should be redirected to the new `pb_institutional_manager` dashboard (currently empty)
5. You should see a new Administer Institutions (instead of networks)
6. Try to visit any super admin url that an institutional manager should not be able to see like modify the appearance of the network
7. Try to edit or modify books that are displayed on the book list
8. Try to assign some books to the institution (as super admin) and you should see them in the book list as institutional manager
9. As a Super admin review the new institution managers are not displayed as network managers on Network Setup
10. * Super Admins menus should remain as always if a superadmin or network manager are logged in
